arch/xtensa/src/esp32/esp32_wlan.c

```diff
@@ -493,6 +493,12 @@ static int wlan_rx_done(void *buffer, uint16_t len, void *eb)
     {
       nwarn("ERROR: Wlan receive %d larger than %d\n",
              len, WLAN_BUF_SIZE);
+
+      if (eb)
+        {
+          esp_wifi_free_eb(eb);
+        }
+
```

Comment:
i guess we should do the same for the above `if (!priv->ifup)` block as well. how do you think?

Reply:
>i guess we should do the same for the above if (!priv->ifup) block as well. how do you think?

I think we should add the logic too.
However, I think I should refactor the logic to make it simple.